ext/dcos-installer/dcos_installer/backend.py

```diff
@@ -5,32 +5,260 @@
+        'bootstrap_url': lambda cloudformation_s3_url: cloudformation_s3_url,
+        'reproducible_artifact_path': calculate_reproducible_artifact_path
+    }, gen.installer.aws.groups['master'][1])
+}
+
+
+aws_advanced_parameters = {
+    'variables': {
+        # TODO(cmaloney): Namespacing would be really handy here...
+        'aws_template_storage_bucket',
+        'aws_template_storage_bucket_path',
+        'aws_template_upload',
+        'aws_template_storage_bucket_path_autocreate',
+        'cloudformation_s3_url',
+        'provider',
+        'bootstrap_url',
+        'bootstrap_variant'
+    },
+    'conditional': {
+        'aws_template_upload': {
+            'true': {
+                'aws_template_storage_access_key_id',
+                'aws_template_storage_secret_access_key',
+                'aws_template_storage_region_name'},
+            'false': {}
+        }
+    }
+}
+
+
+# TODO(cmaloney): Make it so validation happens using the provided AWS credentials.
+def do_aws_cf_configure():
+    """Returns error code
+
+    Generates AWS templates using a custom config.yaml
+    """
+
+    # TODO(cmaloney): Move to Config class introduced in https://github.com/dcos/dcos/pull/623
+    with open(CONFIG_PATH, 'r') as f:
+        config = yaml.load(f)
+
+    aws_config_target = gen.ConfigTarget(aws_advanced_parameters)
+    aws_config_target.add_entry(aws_advanced_entry, False)
+
+    gen_config = stringify_configuration(config)
+    config_targets = [
+        gen.get_dcosconfig_target_and_templates(gen_config, [])[0],
+        aws_config_target]
+
+    messages = gen.validate_config_for_targets(config_targets, gen_config)
+    # TODO(cmaloney): kill this function and make the API return the structured
+    # results api as was always intended rather than the flattened / lossy other
+    # format. This will be an  API incompatible change. The messages format was
+    # specifically so that there wouldn't be this sort of API incompatibility.
+    messages = normalize_config_validation(messages)
+    if messages:
+        print_messages(messages)
+        return 1
+
+    # TODO(cmaloney): This is really hacky but a lot simpler than merging all the config flows into
+    # one currently.
+    # Get out the calculated arguments and manually move critical calculated ones to the gen_config
+    # object.
+    # NOTE: the copying across, as well as validation is guaranteed to succeed because we've already
+    # done a validation run.
+    full_config = gen.calculate_config_for_targets(config_targets, gen_config)
+    gen_config['bootstrap_url'] = full_config['bootstrap_url']
+    gen_config['provider'] = full_config['provider']
+    gen_config['bootstrap_id'] = full_config['bootstrap_id']
+    gen_config['cloudformation_s3_url'] = full_config['cloudformation_s3_url']
+
+    bootstrap_variant = full_config['bootstrap_variant'] if full_config['bootstrap_variant'] else None
```

Comment:
nit: Needs a comment (this is converting from a bootstrap variant string to a bootstrap variant None or string)